overlord/servicestate/servicestate.go

```diff
@@ -42,14 +45,93 @@ import (
 	"github.com/snapcore/snapd/wrappers"
 )
 
+type UserScope struct {
+	Names   []string
+	Keyword bool
+}
+
+// UserList returns a decoded list of users which takes any keyword into account.
+// Takes the current user to be able to handle special keywords like 'user'.
+func (us *UserScope) UserList(currentUser *user.User) ([]string, error) {
+	if !us.Keyword {
+		return us.Names, nil
+	}
+	switch us.Names[0] {
+	case "users":
```

Comment:
we probably want to make the keyword here "all"

Reply:
I updated all usages of "user" and "users" to "self" and "all", also updated spec